### PR TITLE
148 separate emg outputs in hdf5 by dig in

### PR DESCRIPTION
**Pull request by abuzarmahmood**
Update EMG pipeline to allow:

- Simpler handling of trial metadata (taste, laser) using dataframes
- Process emg data without chunking into tastes (simplifies code, and better handles uneven trials)

Includes output of a `trial_info_frame` which is human-readable, and contains all metadata associated with a trial.

### CODE REVIEW

**Comment by abuzarmahmood** [comment]
1. Merge `emg_plot_filt_env` into `emg_filter`
2.

**Comment by abuzarmahmood** [comment]
**All tests passed**

**Review by hfgem — CHANGES_REQUESTED** [review]
See comments - changes needed and new issues needed.

File: emg/emg_freq_setup.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    os.makedirs(plot_dir)
+
+print(f'emg_output_dir: {emg_output_dir}')
+os.chdir(emg_output_dir)  
+
+# Write the emg_env data to a file
+emg_env_df.to_csv('emg_env_df.csv')
+np.save('flat_emg_env_data.npy', flat_emg_env_data)
+
+print('Deleting emg_BSA_results')
```
[the remark this answers is not in the document]
Comment (abuzarmahmood) [reply emg/emg_freq_setup.py:122]: Created an issue for future work

File: emg/gape_QDA_classifier/get_gapes_Li.py [hunk 2]
```diff
@@ -7,56 +7,101 @@
...
+    durations = []
+    final_peak_ind = []
+    for peak in peak_ind:
+        try:
+            left_end = np.where(below_mean_ind < peak)[0][-1]
+            right_end = np.where(below_mean_ind > peak)[0][0]
+        except:
+            continue
+        dur = below_mean_ind[right_end]-below_mean_ind[left_end]
+        if dur > 20.0 and dur <= 200.0:
```
[the remark this answers is not in the document]
Comment (abuzarmahmood) [reply emg/gape_QDA_classifier/get_gapes_Li.py:49]: Agreed, but "technically" this step is a part of the classifier...I didn't bother changing it.

File: emg/gape_QDA_classifier/get_gapes_Li.py [hunk 3]
```diff
@@ -7,56 +7,101 @@
...
-            if i == 0:
-                ax[2*i, j].set_title(taste_names[j])
-            if i == n_laser-1:
-                ax[2*i+1, j].set_xlabel('Time (ms)')
-    return fig, ax
+def QDA_process_single_trial(this_trial, pre_stim, pre_stim_mean):
+
+    peak_ind = detect_peaks(
+        this_trial, 
+        mpd=85,
```
[the remark this answers is not in the document]
Comment (abuzarmahmood) [reply emg/gape_QDA_classifier/get_gapes_Li.py:22]: Same as above comment.

**Review by hfgem — APPROVED** [review]
LGTM

**Pull request merged**